src/bsoncxx/test/bson_builder.cpp

```diff
@@ -1234,4 +1244,526 @@ TEST_CASE("builder::stream::document throws on consecutive keys", "[bsoncxx::bui
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends undefined", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_undefined(&expected, "foo", -1);
+
+    builder::list b{"foo", types::b_undefined{}};
+
+    bson_eq_object(&expected, b.view().get_document().value);
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends oid", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_oid_t oid;
+    bson_oid_init(&oid, NULL);
+
+    bson_append_oid(&expected, "foo", -1, &oid);
+
+    SECTION("b_oid works") {
+        builder::list b{"foo", types::b_oid{bsoncxx::oid{(char*)oid.bytes, 12}}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("raw oid works") {
+        builder::list b{"foo", bsoncxx::oid{(char*)oid.bytes, 12}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends bool", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    SECTION("b_bool true works") {
+        bson_append_bool(&expected, "foo", -1, 1);
+
+        builder::list b{"foo", types::b_bool{true}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("raw true works") {
+        bson_append_bool(&expected, "foo", -1, 1);
+
+        builder::list b{"foo", true};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("b_bool false works") {
+        bson_append_bool(&expected, "foo", -1, 0);
+
+        builder::list b{"foo", types::b_bool{false}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("raw false works") {
+        bson_append_bool(&expected, "foo", -1, 0);
+
+        builder::list b{"foo", false};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends date time", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_date_time(&expected, "foo", -1, 10000);
+
+    builder::list b{"foo", types::b_date{std::chrono::milliseconds{10000}}};
+
+    bson_eq_object(&expected, b.view().get_document().value);
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends null", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_null(&expected, "foo", -1);
+
+    builder::list b{"foo", types::b_null{}};
+
+    bson_eq_object(&expected, b.view().get_document().value);
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends regex", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_regex(&expected, "foo", -1, "^foo|bar$", "i");
+
+    builder::list b{"foo", types::b_regex{"^foo|bar$", "i"}};
```

Comment:
`bson_value` has a constructor for regexes by a pair of strings. Does the following work?

```
builder::list b{"foo", {"^foo|bar$", "i"}};
```

Or perhaps it needs to be made explicit:

```
builder::list b{"foo", bson_value{"^foo|bar$", "i"}};
```

Reply:
Ah that makes sense. Thanks for explaining.